common/flatpak-dir.c

```diff
@@ -4651,6 +4664,9 @@ flatpak_dir_pull (FlatpakDir                           *self,
   /* If @opt_results is set, @opt_rev must be. */
   g_return_val_if_fail (opt_results == NULL || opt_rev != NULL, FALSE);
 
+  /* @repo should either be NULL or a child repo */
+  g_return_val_if_fail (repo == NULL || ostree_repo_equal (ostree_repo_get_parent (repo), self->repo), FALSE);
```

Comment:
This change looks unrelated.

Reply:
If `@repo` is not a child of the directory repo, it doesn't make sense to use it for the `ostree_repo_find_remotes_async` call below. And we do that so we can just consistently use `repo` everywhere rather than `self->repo` in some places, although that shouldn't be strictly necessary.

Reply:
It would be useful to put a summary of your answer from here into the comment, because otherwise it’s not clear why this precondition exists.